### PR TITLE
Auto-generate markdown documentation

### LINKED ISSUE #14: Document commands and options
/kind documentation

We need to have publicly facing documents for `shp`'s commands, sub-commands, and available options.
A lot of this already exists in code - ideally we should be able to generate this documentation and render it in Markdown so it can be consumed by GitHub and the website.

This issue should be focused on having auto-generated reference documentation. A make target should be used to make it easy to generate the docs in CI. A separate make target should verify that any generated doc changes are committed to git on pull requests.

### PR DESCRIPTION
**Pull request by kevydotvinu**
# Changes

Auto-generate markdown documentation using the [cobra library](https://github.com/spf13/cobra/blob/master/doc/md_docs.md).

Fixes #14 

# Submitter Checklist

- [x] ~Includes tests if functionality changed/was added~
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/shp.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```
+shp [command] [resource] [flags]
+```
+
+### Options
+
+```
+      --as string                      Username to impersonate for the operation
+      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
+      --cache-dir string               Default cache directory (default "/home/vkochuku/.kube/cache")
```
Comment (otaviof) [inline docs/shp.md:14]: During the processing of the Cobra commands, the `cache-dir` is being extended and therefore your home folder path is shown as part of the documentation. Please, check how we could have something like `~/.kube/cache` instead.

**Review by otaviof — CHANGES_REQUESTED** [review]
Cool initiative, thanks! Please consider the review comment, that's my only remark.

**Comment by gabemontero** [comment]
/assign @adambkaplan 

since he created the issue being fixed

**Comment by gabemontero** [comment]
/assign @otaviof 

since he provided  a review previously

**Comment by otaviof** [comment]
@kevydotvinu after we finish with the amends, please squash all the changes in two commits. The first with general code changes, and the second with the generated markdown files.

**Comment by kevydotvinu** [comment]
> @kevydotvinu after we finish with the amends, please squash all the changes in two commits. The first with general code changes, and the second with the generated markdown files.

Thanks @otaviof :+1: Appreciate your continuous help :smile: Please review the changes and let me know if anything else to be added/changed.

**Comment by otaviof** [comment]
/lgtm
/approve

Thank you indeed, @kevydotvinu!

**Review by otaviof — APPROVED** [review]

**Pull request merged**